src/extensions/default/InlineBorderRadiusEditor/BorderRadiusEditor.js

```diff
@@ -66,9 +66,14 @@ define(function(require, exports, module) {
         return this.value + (this.value === 0 ? "" : this.unit);
     };
 
-    function BorderRadiusEditor($parent, valueString, radiusChangeHandler) {
+    function BorderRadiusEditor($parent, valueString, radiusChangeHandler, type) {
         var self = this;
-
+        if(type === "border-radius"){
+            BorderRadiusTemplate = require("text!BorderRadiusEditorTemplate.html");
```

Comment:
the templates are basically the same so let's not create two of them.

Instead, try this pattern:
1. Add a function to `BorderRadiusUtils` called `getSingleProperty`. The function will look something like:
```js
function getSingleProperty(property) {
    var property = /top-left|top-right|bottom-left|bottom-right/.exec(property);
    return property && property[0];
}
```
2. Let this function take two extra arguments - one called `isSingleProperty` that is a boolean and another called `iconClassName`. In this function let the default value for the first be `false`. Before calling this function, get the value of `BorderRadiusUtils.getSingleProperty` by passing the css property to it. Store it in a variable called `singleProperty`. Pass the value of `!!singleProperty` as the `isSingleProperty` argument to this function, and pass the value of `singleProperty` as the `iconClassName` argument.
3. In general the idea is to just use the all-corners logic for the single values as well and disable the individual value logic. First, change the template to conditionally render certain parts of the template. To do that, wrap each part that needs to be conditionally rendered in a [mustache section](https://github.com/janl/mustache.js#sections). For e.g., do this in the template
```html
{{#individualCorners}}
<div id="individualCornerArea">
    ...
</div>
{{/individualCorners}}
```
and
```html
{{#inlineEditorHeader}}
<header>
    <ul class="button-bar">
        ...
    </ul>
</header>
{{/inlineEditorHeader}}
```
and
```html
{{#allCornerIcon}}
<div class="all corner-icon">
    <div class="dot"></div>
    <div class="dot"></div>
    <div class="dot"></div>
    <div class="dot"></div>
</div>
{{/allCornerIcon}}
{{#singleCornerIcon}}
<div class="{{iconClassName}} corner-icon"></div>
{{/singleCornerIcon}}
```
4. Then in this function, you modify the value of strings to determine what to render. Your code will look something like this:
```js
var values = JSON.parse(JSON.stringify(Strings));

if (!isSingleProperty) {
    values["inlineEditorHeader"] = true;
    values["individualCorners"] = true;
    values["allCornerIcon"] = true;
} else {
    values["singleCornerIcon"] = true;
}

self.$element = $(Mustache.render(BorderRadiusTemplate, values));
```
The rest of the logic should handle itself the way it is. Try it out and feel free to ask questions.

Reply:
Thank you so much @gideonthomas for such awesome guidance. I'll try this one out.